pkg/cluster/cluster.go

```diff
@@ -262,16 +260,25 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 				continue // Back-off, either on normal recovery or error.
 			}
 
-			if err := c.reconcile(running); err != nil {
-				c.logger.Errorf("fail to reconcile: %v", err)
-				if isFatalError(err) {
-					c.logger.Errorf("exiting for fatal error: %v", err)
+			// TODO: The case "c.members = nil" only happens on creating seed member.
+			//       We should just set the member directly if successful.
+			if rerr != nil || c.members == nil {
```

Comment:
It would be easier to maintain if we could centralize the logic. Even if we want to optimize it to do less member listing, we also need to centralized point for those member stuff and check error there.